langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/ToExpString.java

```diff
@@ -0,0 +1,114 @@
+
+/**
+ * Native implementation of lang.float:toExpString(float, int?).
+ *
+ * @since 2.0.0
+ */
+public class ToExpString {
+    public static BString toExpString(double x, Object fractionDigits) {
+        // If `x` is NaN or infinite, the result will be the same as `value:toString`.
+        if (Double.isInfinite(x) || Double.isNaN(x)) {
+            return StringUtils.fromString(StringUtils.getStringValue(x, null));
+        }
+
+        long noOfFractionDigits;
+
+        // If fractionalDigits is `()`, use the minimum number of digits required to accurately represent the value.
+        double xAbsValue = Math.abs(x);
+        if (fractionDigits == null) {
+            int integerPart = (int) (Math.log10(xAbsValue));
+            noOfFractionDigits = BigDecimal.valueOf(xAbsValue / Math.pow(10, integerPart)).scale();
+        } else {
+            noOfFractionDigits = (long) fractionDigits;
+        }
+
+        // Panic if noOfFractionDigits < 0.
+        if (noOfFractionDigits < 0) {
+            throw ErrorCreator.createError(getModulePrefixedReason(FLOAT_LANG_LIB,
+                    BallerinaErrorReasons.INVALID_FRACTION_DIGITS_ERROR),
+                    BLangExceptionHelper.getErrorDetails(RuntimeErrors.INVALID_FRACTION_DIGITS));
+        }
+
+        if (noOfFractionDigits > Integer.MAX_VALUE) {
+            // Maximum value a double can represent is around 1.7*10^308.
+            noOfFractionDigits = 308;
+        }
+
+        int exponent = (int) noOfFractionDigits;
+
+        int tens = 0;
+        if (xAbsValue != 0 && xAbsValue < 1) {
+            double multipliedValue = xAbsValue;
+
+            while (multipliedValue < 1) {
+                multipliedValue = xAbsValue * Math.pow(10, tens + 1);
+                tens++;
+            }
+        }
+
+        DecimalFormat decimalFormat = new DecimalFormat();
+        decimalFormat.setRoundingMode(RoundingMode.HALF_EVEN);
+
+
+        BigDecimal numberBigDecimal = new BigDecimal(x);
+        if (xAbsValue != 0 && xAbsValue < 1) {
+            numberBigDecimal  = numberBigDecimal.setScale(exponent + tens, RoundingMode.HALF_EVEN);
+
+        } else {
+            numberBigDecimal  = numberBigDecimal.setScale(exponent, RoundingMode.HALF_EVEN);
+        }
+
+        String power = "0".repeat(exponent);
+
+        decimalFormat = new DecimalFormat("0." + power + "E0");
+        String res = decimalFormat.format(numberBigDecimal);
+        int indexOfExp = res.lastIndexOf("E");
+        String firstSection = res.substring(0, indexOfExp);
+        int idxOfDecimalPoint = firstSection.lastIndexOf(".");
+        if (idxOfDecimalPoint == firstSection.length() - 1) {
+            firstSection = res.substring(0, idxOfDecimalPoint);
+        }
+        String secondSection = res.substring(indexOfExp + 1);
+        int p = Integer.parseInt(secondSection);
+        if (p >= 0) {
+            secondSection = "e+" + secondSection;
```

Comment:
IMO this is wrong simple `e` means natural exponent right...

Reply:
The spec says `e`.
https://github.com/ballerina-platform/ballerina-spec/blob/master/lang/lib/float.bal#L298